### PR TITLE
Get rid of lib to merge tools v2.13.1

### LINKED ISSUE #356: Get template v2.13.1 (get rid of lib)
### Description of feature

Follow the example [here](https://nf-co.re/blog/2024/tools-2_13)

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Closes #356

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

File: result/pipeline_info/params_2024-03-19_16-59-36.json
Comment (bjlang) [inline result/pipeline_info/params_2024-03-19_16-59-36.json:1]: I believe this file can be deleted

File: workflows/atacseq.nf [hunk 1]
```diff
@@ -711,27 +681,29 @@ workflow ATACSEQ {
...
 
     //
-    // MODULE: Pipeline reporting
+    // Collate and save software versions
     //
-    CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
-    )
+    softwareVersionsToYAML(ch_versions)
+        .collectFile(storeDir: "${params.outdir}/pipeline_info", name: 'nf_core_rnaseq_software_mqc_versions.yml', sort: true, newLine: true)
```
Comment (bjlang) [inline workflows/atacseq.nf]: change name to nf_core_atacseq_software_mqc_versions.yml

File: lib/WorkflowAtacseq.groovy [hunk 2]
```diff
@@ -1,160 +0,0 @@
...
-
-        if (!params.macs_gsize) {
-            macsGsizeWarn(log)
-        }
-
-        if (!params.read_length && !params.macs_gsize) {
-            log.error "Both '--read_length' and '--macs_gsize' not specified! Please specify either to infer MACS2 genome size for peak calling."
-            System.exit(1)
-        }
-    }
```
Comment (bjlang) [inline lib/WorkflowAtacseq.groovy:37]: Couldn't find these 2 checks in the new version anymore

File: subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    }
+
+    // Check that multiple runs of the same sample are of the same datatype i.e. single-end / paired-end
+    def endedness_ok = metas.collect{ it.single_end }.unique().size == 1
+    if (!endedness_ok) {
+        error("Please check input samplesheet -> Multiple runs of a sample must be of the same datatype i.e. single-end or paired-end: ${metas[0].id}")
+    }
+
+    return [ metas[0], fastqs ]
+}
```
Comment (bjlang) [inline subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf]: I don't see where this code is coming from and also doubt its correctness. We don't have a strandedness defined in the samplesheet. The single_end check makes sense but is imo already included in the check_samplesheet.py. Furthermore, I don't see this function being called anywhere.

File: subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    // Uncomment function in methodsDescriptionText to render in MultiQC report
+    def reference_text = [
+            "<li>Andrews S, (2010) FastQC, URL: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/).</li>",
+            "<li>Ewels, P., Magnusson, M., Lundin, S., & Käller, M. (2016). MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics , 32(19), 3047–3048. doi: /10.1093/bioinformatics/btw354</li>"
+        ].join(' ').trim()
+
+    return reference_text
+}
+
+def methodsDescriptionText(mqc_methods_yaml) {
```
Comment (bjlang) [inline subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf:224]: Where is this function used?

File: main.nf [hunk 5]
```diff
@@ -13,74 +13,139 @@ nextflow.enable.dsl = 2
...
+        params.bowtie2_index,
+        params.chromap_index,
+        params.star_index,
+        params.macs_gsize,
+        params.read_length
+    )
+    ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)
+
+    //
+    // WORKFLOW: Run nf-core/rnaseq workflow
```
Comment (bjlang) [inline main.nf]: ATACseq :)

File: workflows/atacseq.nf [hunk 6]
```diff
@@ -776,34 +748,12 @@ workflow ATACSEQ {
...
-    }
-
-    NfcoreTemplate.summary(workflow, params, log)
-}
-
-workflow.onError {
-    if (workflow.errorReport.contains("Process requirement exceeds available memory")) {
-        println("🛑 Default resources exceed availability 🛑 ")
-        println("💡 See here on how to configure pipeline: https://nf-co.re/docs/usage/configuration#tuning-workflow-resources 💡")
-    }
```
Comment (bjlang) [inline workflows/atacseq.nf:806]: I haven't found this anymore

**Review by bjlang — CHANGES_REQUESTED** [review]

File: workflows/atacseq.nf (hunk 1 above)
Comment (JoseEspinosa) [reply workflows/atacseq.nf]: Addressed [here](https://github.com/nf-core/atacseq/pull/357)

File: result/pipeline_info/params_2024-03-19_16-59-36.json
Comment (JoseEspinosa) [reply result/pipeline_info/params_2024-03-19_16-59-36.json:1]: Deleted

File: main.nf (hunk 5 above)
Comment (JoseEspinosa) [reply main.nf]: Fixed [here](https://github.com/nf-core/atacseq/blob/eb1285ae0a78b67912f96f5fed3a7ff5ac2a8e31/main.nf#L83)

File: workflows/atacseq.nf (hunk 6 above)
Comment (JoseEspinosa) [reply workflows/atacseq.nf:806]: Added the updated version used in rnaseq [here](https://github.com/nf-core/atacseq/blob/eb1285ae0a78b67912f96f5fed3a7ff5ac2a8e31/subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf#L115-L118)

File: subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf (hunk 3 above)
Comment (JoseEspinosa) [reply subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf]: 👌 Removed

File: lib/WorkflowAtacseq.groovy (hunk 2 above)
Comment (JoseEspinosa) [reply lib/WorkflowAtacseq.groovy:37]: Added again [here](https://github.com/nf-core/atacseq/blob/eb1285ae0a78b67912f96f5fed3a7ff5ac2a8e31/subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf#L145-L152)

File: subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf (hunk 4 above)
Comment (JoseEspinosa) [reply subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf:224]: This is added by the template to make possible to dump the reference e.g. [here](https://github.com/nf-core/atacseq/blob/eb1285ae0a78b67912f96f5fed3a7ff5ac2a8e31/subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf#L199-L201).

Comment (JoseEspinosa) [reply subworkflows/local/utils_nfcore_atacseq_pipeline/main.nf:224]: And yes I think we can leave this to the template merge

**Review by bjlang — APPROVED** [review]
LGTM

**Pull request merged**